addon/wolfcrypt/h/ecc.h

```diff
@@ -19,7 +19,9 @@
  * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1335, USA
  */
 #include <napi.h>
-#include "wolfssl/options.h"
```

Comment:
Everything looks good and these are fine but they've already been added as a part of https://github.com/wolfSSL/wolfcrypt_nodejs/pull/7. Would like to have Kaleb review the FIPS related stuff of my PR then merge this after

Reply:
Understood. I've subscribed to notifications on that PR & will revisit / rebase once it is merged.

Reply:
@jpbland1 I've refreshed & squashed from upstream. 

Using the [user_settings.h](https://github.com/gojimmypi/wolfssl/blob/win-dev/IDE/WIN/user_settings.h) from my [win-dev](https://github.com/gojimmypi/wolfssl/tree/win-dev), in particular with this extra section:

```
/* npm */
#define NPM_WOLFCRYPT
#ifdef NPM_WOLFCRYPT
    /* Optional debug */
    /* #define DEBUG_WOLFSSL */

    /* Optional RNG */
    /* #define WC_RNG_SEED_CB */

    #define HAVE_PKCS7
    #define HAVE_AES_KEYWRAP
    #define WOLFSSL_AES_DIRECT
    #define HAVE_X963_KDF
    #define WOLFSSL_SHA224
    #define WOLFSSL_KEY_GEN
    #define HAVE_ECC
    #define ECC_MAX_BITS 521
    #define WC_ECC256
    #define WC_ECC384
    #define WC_ECC521
    #define HAVE_ECC_ENCRYPT
    #define WOLFSSL_UINT128_T_DEFINED
    #define WOLFSSL_SHA512
    #define WOLFSSL_SHA384
    #define WOLFSSL_SHA3

    #define NO_OLD_RNGNAME
    #define TFM_TIMING_RESISTANT
    #define ECC_TIMING_RESISTANT
    #define WC_RSA_BLINDING
    #define TFM_ECC256
    #define ECC_SHAMIR
    #define ECC_MIN_KEY_SZ 224
    #define HAVE_ECC_BRAINPOOL
    #define HAVE_CURVE25519
    #define FP_ECC
    #define HAVE_ECC_ENCRYPT
    #define WOLFCRYPT_HAVE_ECCSI
    #define WOLFSSL_CUSTOM_CURVES
#endif
```

Running `etup_env.ps1` from a powershell prompt:

```
C:\workspace\wolfcrypt_nodejs-gojimmypi-pr [pr-windows-support ≡ +1 ~0 -0 !]> .\setup_env.ps1
Found wolfSSL at C:/workspace/wolfssl-gojimmypi
WOLFSSL_LIB_PATH:           C:/workspace/wolfssl-gojimmypi/DLL Release/x64
WOLFSSL_INCLUDE_PATH:       C:/workspace/wolfssl-gojimmypi
WOLFSSL_USER_SETTINGS_PATH: C:/workspace/wolfssl-gojimmypi/IDE/WIN
Using Node v20.18.0
v20.18.0
10.8.2
Launching Visual Studio 2022 Community at: C:\Program Files\Microsoft Visual Studio\2022\Community\Common7\IDE\devenv.exe
Clean...

> wolfcrypt@1.0.3 clean
> node-gyp clean

gyp info it worked if it ends with ok
gyp info using node-gyp@9.4.1
gyp info using node@20.18.0 | win32 | x64
gyp info ok
gyp info it worked if it ends with ok
gyp info using node-gyp@9.4.1
gyp info using node@20.18.0 | win32 | x64
gyp info ok
Rebuild...
gyp info it worked if it ends with ok
gyp info using node-gyp@9.4.1
gyp info using node@20.18.0 | win32 | x64
gyp info find Python using Python version 3.8.2 found at "C:\Users\gojimmypi\AppData\Local\Programs\Python\Python38\python.exe"
gyp info find VS using VS2022 (17.11.35327.3) found at:
gyp info find VS "C:\Program Files\Microsoft Visual Studio\2022\Community"
gyp info find VS run with --verbose for detailed information
gyp info spawn C:\Users\gojimmypi\AppData\Local\Programs\Python\Python38\python.exe
gyp info spawn args [
gyp info spawn args   'C:\\workspace\\wolfcrypt_nodejs-gojimmypi-pr\\node_modules\\node-gyp\\gyp\\gyp_main.py',
gyp info spawn args   'binding.gyp',
gyp info spawn args   '-f',
gyp info spawn args   'msvs',
gyp info spawn args   '-I',
gyp info spawn args   'C:\\workspace\\wolfcrypt_nodejs-gojimmypi-pr\\build\\config.gypi',
gyp info spawn args   '-I',
gyp info spawn args   'C:\\workspace\\wolfcrypt_nodejs-gojimmypi-pr\\node_modules\\node-gyp\\addon.gypi',
gyp info spawn args   '-I',
gyp info spawn args   'C:\\Users\\gojimmypi\\AppData\\Local\\node-gyp\\Cache\\20.18.0\\include\\node\\common.gypi',
gyp info spawn args   '-Dlibrary=shared_library',
gyp info spawn args   '-Dvisibility=default',
gyp info spawn args   '-Dnode_root_dir=C:\\Users\\gojimmypi\\AppData\\Local\\node-gyp\\Cache\\20.18.0',
gyp info spawn args   '-Dnode_gyp_dir=C:\\workspace\\wolfcrypt_nodejs-gojimmypi-pr\\node_modules\\node-gyp',
gyp info spawn args   '-Dnode_lib_file=C:\\\\Users\\\\gojimmypi\\\\AppData\\\\Local\\\\node-gyp\\\\Cache\\\\20.18.0\\\\<(target_arch)\\\\node.lib',
gyp info spawn args   '-Dmodule_root_dir=C:\\workspace\\wolfcrypt_nodejs-gojimmypi-pr',
gyp info spawn args   '-Dnode_engine=v8',
gyp info spawn args   '--depth=.',
gyp info spawn args   '--no-parallel',
gyp info spawn args   '--generator-output',
gyp info spawn args   'C:\\workspace\\wolfcrypt_nodejs-gojimmypi-pr\\build',
gyp info spawn args   '-Goutput_dir=.'
gyp info spawn args ]
gyp info spawn C:\Program Files\Microsoft Visual Studio\2022\Community\MSBuild\Current\Bin\MSBuild.exe
gyp info spawn args [
gyp info spawn args   'build/binding.sln',
gyp info spawn args   '/clp:Verbosity=minimal',
gyp info spawn args   '/nologo',
gyp info spawn args   '/p:Configuration=Release;Platform=x64'
gyp info spawn args ]

  nothing.c
  win_delay_load_hook.cc
  nothing.vcxproj -> C:\workspace\wolfcrypt_nodejs-gojimmypi-pr\build\Release\\nothing.lib
  main.cpp
  evp.cpp
  hmac.cpp
  rsa.cpp
  sha.cpp
  ecc.cpp
  pbkdf2.cpp
  pkcs7.cpp
  pkcs12.cpp
  win_delay_load_hook.cc
     Creating library C:\workspace\wolfcrypt_nodejs-gojimmypi-pr\build\Release\wolfcrypt.lib and object C:\workspace\wo
  lfcrypt_nodejs-gojimmypi-pr\build\Release\wolfcrypt.exp
  Generating code
  Previous IPDB not found, fall back to full compilation.
  All 443 functions were compiled because no usable IPDB/IOBJ from previous compilation was found.
  Finished generating code
  wolfcrypt.vcxproj -> C:\workspace\wolfcrypt_nodejs-gojimmypi-pr\build\Release\\wolfcrypt.node
gyp info ok
Install...

> wolfcrypt@1.0.3 install
> node-gyp rebuild

gyp info it worked if it ends with ok
gyp info using node-gyp@9.4.1
gyp info using node@20.18.0 | win32 | x64
gyp info find Python using Python version 3.8.2 found at "C:\Users\gojimmypi\AppData\Local\Programs\Python\Python38\python.exe"
gyp info find VS using VS2022 (17.11.35327.3) found at:
gyp info find VS "C:\Program Files\Microsoft Visual Studio\2022\Community"
gyp info find VS run with --verbose for detailed information
gyp info spawn C:\Users\gojimmypi\AppData\Local\Programs\Python\Python38\python.exe
gyp info spawn args [
gyp info spawn args   'C:\\workspace\\wolfcrypt_nodejs-gojimmypi-pr\\node_modules\\node-gyp\\gyp\\gyp_main.py',
gyp info spawn args   'binding.gyp',
gyp info spawn args   '-f',
gyp info spawn args   'msvs',
gyp info spawn args   '-I',
gyp info spawn args   'C:\\workspace\\wolfcrypt_nodejs-gojimmypi-pr\\build\\config.gypi',
gyp info spawn args   '-I',
gyp info spawn args   'C:\\workspace\\wolfcrypt_nodejs-gojimmypi-pr\\node_modules\\node-gyp\\addon.gypi',
gyp info spawn args   '-I',
gyp info spawn args   'C:\\Users\\gojimmypi\\AppData\\Local\\node-gyp\\Cache\\20.18.0\\include\\node\\common.gypi',
gyp info spawn args   '-Dlibrary=shared_library',
gyp info spawn args   '-Dvisibility=default',
gyp info spawn args   '-Dnode_root_dir=C:\\Users\\gojimmypi\\AppData\\Local\\node-gyp\\Cache\\20.18.0',
gyp info spawn args   '-Dnode_gyp_dir=C:\\workspace\\wolfcrypt_nodejs-gojimmypi-pr\\node_modules\\node-gyp',
gyp info spawn args   '-Dnode_lib_file=C:\\\\Users\\\\gojimmypi\\\\AppData\\\\Local\\\\node-gyp\\\\Cache\\\\20.18.0\\\\<(target_arch)\\\\node.lib',
gyp info spawn args   '-Dmodule_root_dir=C:\\workspace\\wolfcrypt_nodejs-gojimmypi-pr',
gyp info spawn args   '-Dnode_engine=v8',
gyp info spawn args   '--depth=.',
gyp info spawn args   '--no-parallel',
gyp info spawn args   '--generator-output',
gyp info spawn args   'C:\\workspace\\wolfcrypt_nodejs-gojimmypi-pr\\build',
gyp info spawn args   '-Goutput_dir=.'
gyp info spawn args ]
gyp info spawn C:\Program Files\Microsoft Visual Studio\2022\Community\MSBuild\Current\Bin\MSBuild.exe
gyp info spawn args [
gyp info spawn args   'build/binding.sln',
gyp info spawn args   '/clp:Verbosity=minimal',
gyp info spawn args   '/nologo',
gyp info spawn args   '/p:Configuration=Release;Platform=x64'
gyp info spawn args ]

  nothing.c
  win_delay_load_hook.cc
  nothing.vcxproj -> C:\workspace\wolfcrypt_nodejs-gojimmypi-pr\build\Release\\nothing.lib
  main.cpp
  evp.cpp
  hmac.cpp
  rsa.cpp
  sha.cpp
  ecc.cpp
  pbkdf2.cpp
  pkcs7.cpp
  pkcs12.cpp
  win_delay_load_hook.cc
     Creating library C:\workspace\wolfcrypt_nodejs-gojimmypi-pr\build\Release\wolfcrypt.lib and object C:\workspace\wo
  lfcrypt_nodejs-gojimmypi-pr\build\Release\wolfcrypt.exp
  Generating code
  Previous IPDB not found, fall back to full compilation.
  All 443 functions were compiled because no usable IPDB/IOBJ from previous compilation was found.
  Finished generating code
  wolfcrypt.vcxproj -> C:\workspace\wolfcrypt_nodejs-gojimmypi-pr\build\Release\\wolfcrypt.node
gyp info ok

up to date, audited 100 packages in 27s

5 packages are looking for funding
  run `npm fund` for details

found 0 vulnerabilities
Test...

> wolfcrypt@1.0.3 test
> node test.js

Running test:  ecc_makeKey
wc_ecc_make_key...PASS ecc make_key

Running test:  ecc_sharedSecret32
PASS ecc sharedSecret32

Running test:  ecc_sharedSecret64
PASS ecc sharedSecret64

Running test:  ecc_signVerify
wc_ecc_make_key...PASS ecc signVerify

Running test:  ecc_importExportx963
wc_ecc_make_key...PASS ecc importExportx963

Running test:  ecc_importExportDer
wc_ecc_make_key...PASS ecc importExportDer

Running test:  evp_encrypt
PASS evp encrypt

Running test:  evp_decrypt
PASS evp decrypt

Running test:  evp_encryptDecryptOdd
PASS evp encrypt_decrypt_odd

Running test:  evp_encryptionStream
PASS evp encryptionStream

Running test:  evp_decryptionStream
PASS evp decryptionStream

Running test:  evp_encryptDecryptPipes
PASS evp encryptDecryptPipes

Running test:  hmac
PASS hmac hmac

Running test:  hmacStream
PASS hmac hmacStream

Running test:  hmacPipe
PASS hmac hmacPipe

Running test:  pbkdf2
PASS pbkdf2

Running test:  createAndParse
PASS PKCS12 createAndParse

Running test:  pkcs7_addCertificate
PASS pkcs7 addCertificate

Running test:  pkcs7_encodeData
PASS pkcs7 encodeData

Running test:  pkcs7_signVerify
PASS pkcs7 signVerify

Running test:  pkcs7_getAttribute
PASS pkcs7 getAttribute

Running test:  pkcs7_getSid
PASS pkcs7 getSid

Running test:  rsa_MakeRsaKey
PASS rsa MakeRsaKey

Running test:  rsa_keyToDer
PASS rsa keyToDer

Running test:  rsa_keyToPublicDer
PASS rsa keyToPublicDer

Running test:  rsa_privateKeyDecode
PASS rsa privateKeyDecode

Running test:  rsa_publicKeyDecode
PASS rsa publicKeyDecode

Running test:  rsa_encryptDecrypt
PASS rsa encryptDecrypt

Running test:  rsa_signVerify
PASS rsa signVerify

Running test:  sha
PASS sha sha

Running test:  sha224
PASS sha sha224

Running test:  sha256
PASS sha sha256

Running test:  sha384
PASS sha sha384

Running test:  sha512
PASS sha sha512

Running test:  sha512_224
PASS sha sha512_224

Running test:  sha512_256
PASS sha sha512_256

Confirmed reference ./lib/user_settings.h matches wolfSSL source.
```